lib/bundles/inspec-compliance/api.rb

```diff
@@ -6,12 +6,22 @@
 require 'uri'
 
 module Compliance
+  class ServerConfigurationMissing < StandardError
+  end
+
   # API Implementation does not hold any state by itself,
   # everything will be stored in local Configuration store
   class API # rubocop:disable Metrics/ClassLength
     # return all compliance profiles available for the user
     def self.profiles(config)
-      config['server_type'] == 'automate' ? url = "#{config['server']}/#{config['user']}" : url = "#{config['server']}/user/compliance"
+      # Chef Compliance
+      if is_compliance_server?(config)
+        url = "#{config['server']}/user/compliance"
+      # Chef Automate
+      elsif is_automate_server?(config)
+        url = "#{config['server']}/profiles/#{config['user']}"
+      end
```

Comment:
Let's catch the third case i.e. `else raise ...` becuse there is no url instead of getting the nil error further down